pkg/kubelet/pleg/generic.go

```diff
@@ -444,9 +444,24 @@ func updateRunningPodAndContainerMetrics(pods []*kubecontainer.Pod) {
 			}
 		}
 	}
+
 	for key, value := range containerStateCount {
 		metrics.RunningContainerCount.WithLabelValues(key).Set(float64(value))
 	}
+	if metrics.RunningContainerCount.IsCreated() {
+		// reset runningContainerCount to avoid stale metrics.
+		// For example, in last call, there's 1 container in unknown state, then the container's state changed to running
+		// state in current call, RunningContainerCount metrics should not keep 1 container in unknown state.
+		for _, state := range []kubecontainer.State{
+			kubecontainer.ContainerStateRunning,
+			kubecontainer.ContainerStateCreated,
+			kubecontainer.ContainerStateExited,
+			kubecontainer.ContainerStateUnknown} {
+			if _, ok := containerStateCount[string(state)]; !ok {
+				metrics.RunningContainerCount.DeleteLabelValues(string(state))
```

Comment:
The PR description and this solution don't seem to align. Don't you want to reset only the unknown states?

Reply:
No, this pr checks all container states, not only containers that is created, but also containers in `Exited` state or other states.   Simply to say, when `containerStateCount` has no items, that is to say, no `Running`, `Exited`, `Created`, `Unknown` containers,  all values set in `metrics.RunningContainerCount` should be deleted.

Reply:
You are right, pr description is wrong. Initially,  `metrics.RunningContainerCount` is reset before the `for key,value := range containerStateCount` loop, but that solution will raise a race while at the same time metric collector is collecting this metric.  Then, I changed the logic to remove values in `metrics.RunningContainerCount` that is not in `containerStateCount` in case to cause wrong metric values.

Reply:
Ah, I misunderstood. Yeah then this makes sense to me.

Reply:
/lgtm

(From a metrics perspective)

Reply:
Thank you very much. By the way, is any docs explains how to make this pr approved or merged?  Should I add milestone for this pr?

Reply:
You still need an lgtm from a node approver. 

/cc @dashpole @ehashman

Reply:
Thanks a lot.